### ISSUE TITLE
New chat features, user info, offline, monitoring

### ISSUE THREAD

**Issue by joscdk** [zammad/zammad#1407]
Delete this if you must, some of these features is already suggested as features, and some isn't, just wanted to add our needs :)

## Feature list

- Ask for user name, email & message before starting chat
- Allow users to create ticket through chat widget even when offline
- Show users online on the page from the chat, and when a user start the chat, you can see their movement
- Allow for pre filled name and email

### 1. Ask for user name, email, message

Should be more or less explanatory

### 2. Offline ticket

When the chat is offline, and a user try to start a chat, it will be created as a ticket

### 3. Monitoring

Show users online on the page. This helps giving a short user overview. If a user starts a chat, you can see their movement before they start the chat, can help debug in various cases.

### 4. Pre filled name/email

This relates to the first, allow for entering name and email in the Javascript integration, if the chat is behind a login

Zammad is really a great system, and i think the above features would be nice features. Especially 1, 2 & 4

**Comment by hanneshal** [comment]
Ok, nice wish-list.
One question: 

> Allow for pre filled name and email
Where do we get the name and email from? Maybe I don't understand the scenario right now. But there is no internal chat for users or customer. So this has to be an "external" non-Zammad system. Or did I miss something?
I think if you would implement:
- name
- email 
- message 
before starting the chat (which we dismissed on purpose btw) then it would be easy to set those values via JS from the external application. 
Maybe you can explain your use case.

Thanks

Johannes

**Comment by joscdk** [comment]
In our case, with our current chat system, we have it implemented 2 places:

1: Our main website.
Here people open the chat, fill out name, email and message. Then it starts the chat, and we already have their details.

2. In our dashboard
This is where our clients use our system, and it's for support. We fill in name and email from our user database.
Then people don't need to fill it out, and we know who the user is, making it easier to help them :)
Basically the data is set in the JS integration. Of cause, if the form is there already, it's possible to set it through JS :)

**Comment by hanneshal** [comment]
OK, thank you

**Comment by semaf** [comment]
#1227 Also some feature requests

**Comment by ammo8600** [comment]
Has anything been added so far?

**Comment by xeboo** [comment]
Unfortunately, these wishes have not yet been implemented :( I wait for it from version to version

**Comment by MrGeneration** [comment]
This is in our backlog, we never promised to implement this too soon. :-x 
Sorry about that. Currently we're focussing on knowledge base and fixing other issues and enhancing existend functions.

I'll be locking the conversation of this issue for now, to reduce the nice on the repository (it's nothing personal, but on each comment over 1400 people receive a notification). If you want to discuss about this, feel free to do so on our community: https://community.zammad.org/
